cloudsync/api_test.py

```diff
@@ -191,35 +192,46 @@ def test_process_transcode_results(mocker):
     assert len(video.videothumbnail_set.all()) == 1
 
 
-def test_extract_title():
+@pytest.mark.parametrize(
+    'course_prefix, session, date_str, expected_record_date', [
+        ('MIT-6.046-2017-Spring', 'L01', '2017apr06', datetime(2017, 4, 6)),
+        ('abcdefg', 'L01', '2017apr06', datetime(2017, 4, 6)),
+        ('MIT-6.046-2017-Spring', None, '2017apr06', datetime(2017, 4, 6)),
+        ('/&*3:<>俺正和', None, '2017apr06', datetime(2017, 4, 6)),
+        ('abcdefg', None, '2017badmonthvalue06', None),
```

Comment:
Add another test case for filenames that have a room number instead of a session, such as `MIT-Adhoc-lec-mit-0000-2017aug10-0954-2-190.mp4`:

`('MIT-Adhoc', '2-190', '2017aug10', datetime(2017, 8, 10))`